### ISSUE TITLE
Midpoint Review - Accessibility Feedback - Benefits Non-Disability, Burial 530, Form Updates

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#72433]
## VFS actions
- [x] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] Comment on this ticket:
  - [ ] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [ ] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [ ] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] Close the ticket when all feedback has been addressed.

## Thoughts/questions

- Thank you for such a thorough prototype! It's not often that we get all the error states at a midpoint, and I'm very grateful for that level of effort.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- **Should:** There are a few places where you're conditionally revealing follow-up questions. Jeana Clark shared with me the rules that she's going to propose for the design system, which I think are solid (even if they're not official yet):
   
   > Conditionally revealed fields can be used if the following conditions are met:
   > 1. There should only be one reveal on a page.
   > 2. When the revealed trigger is selected, you must be able to tab directly into the newly revealed field. (which is why we've put the "other" question last.)
   > 3. The newly revealed question field must be understood by itself.  For example, don't just say "Other". Say, "Since your relationship with the veteran was not listed, please describe it here"
   
   These all ultimately amount to "keep the interaction as simple as possible." For screen reader users that don't get any visual cues about what's changed on the page, it can get confusing if it feels like things are changing on the page without a cause-and-effect that's easy to figure out.
   
   Providing all of that as background and as a way to keep myself honest with the recommendations I'm about to give!
   
   - [Step 2, Burial information](https://www.sketch.com/s/de782a35-e147-4c32-a2a8-ba53071ec8e7/a/mP0pKYw) - The warning alert appears when a user enters dates greater than 2 years from the current date. Since this isn't revealing a new question that can be tabbed to, there's extra focus management work that would need to be done to ensure screen readers announce the alert. I think this could be better handled by moving the warning text into the instructions for the page, and/or using normal error validation when the user presses Continue.
   - [Step 3, Previous name](https://www.sketch.com/s/de782a35-e147-4c32-a2a8-ba53071ec8e7/a/ZVRlJnv) - It looks like a lot of extra fields are displayed when you select "Yes." Since users would have to tab past "No" to get to them, and there are multiple new fields, this is a good candidate for moving to a new page.
   - [Step 4, Type of burial allowance](https://www.sketch.com/s/de782a35-e147-4c32-a2a8-ba53071ec8e7/a/qb0vbgy) - I think this is okay as a conditionally-revealed field and it meets Jeana's criteria above, but there's enough helper text that's being revealed that I think it's worth considering moving to another page.
   - [Step 4, Plot or interment allowance](https://www.sketch.com/s/de782a35-e147-4c32-a2a8-ba53071ec8e7/a/5n4qnvy) - Selecting "Yes" is what reveals the new field, which means users will have to tab past the "No" option before getting to the new field. Better on a new page, or possibly there's a way you can change the grammar of the question to make it work here.
   - [Step 4, Plot or interment allowance](https://www.sketch.com/s/de782a35-e147-4c32-a2a8-ba53071ec8e7/a/Ee95eVo) - Same as above -- triggered by "Yes" option, so not a direct tab to the new field.
   - [Step 4, Plot or interment allowance](https://www.sketch.com/s/de782a35-e147-4c32-a2a8-ba53071ec8e7/a/r90498D) - Two responses trigger the conditional reveal, and both reveal multiple new fields. I think these are more appropriate on a new page.
- **Should:** When each step is broken into multiple pages, there's an H3 on each page describing the sub-step you're on. Where possible, these H3s should be unique, at least within their step. For example, Step 4 has multiple pages with an H3 of "Plot or interment allowance." Especially for screen reader users, this makes it a little harder to parse if you've actually advanced to a new page.
- **Must:** On [Step 4, Costs you paid for](https://www.sketch.com/s/de782a35-e147-4c32-a2a8-ba53071ec8e7/a/VEY5EJw), there's helper text that comes after the checkboxes clarifying the requirements for checking those boxes. Two potential issues here:
   - Many screen reader users are likely to press [tab] after the checkboxes to jump to the next focusable element, either the next form field or the back/continue buttons. That means they're likely to skip over this text without ever hearing it.
   - The text seems important enough that users should encounter it before they make their selections with the checkboxes, not after (and then have to backtrack when they realize they made a mistake).
   
   My recommendation here is to put that helper text before the checkboxes, and to code it as part of the `legend` (styled to make it look like regular text):
   ```
   <fieldset>
      <legend>
         <h3 class="some-class">Costs you paid for</h3>
         <span class="some-other-class">By selecting any of these costs...</span>
      </legend>
      <va-checkbox-group />
   </fieldset>
   ```
   
   It's not the most elegant code to stick a bunch of extra text into the `legend`, but doing that ensures all of that text gets announced to screen reader users when they tab to the first checkbox. If it's not in the `legend`, it's likely to be skipped over.
- **Must:** Similar to the above, [Step 4, Type of burial allowance] has a checkbox with a lot of helper text. You'll want to use a similar pattern with the helper text bundled in with a `legend` to ensure screen readers read it when tabbing to the checkbox.
- **Consider:** There are a couple of places where terms are used that I found a little confusing or unexpected, where it might be worth using the additional info component to provide definitions or context. This is all up to you, since you have a better sense of who's filling out these forms than I do! But the ones that stood out to me were:
   - [Step 3, Previous name](https://www.sketch.com/s/de782a35-e147-4c32-a2a8-ba53071ec8e7/a/ZVRlJnv) - "Services rendered under that name" isn't a common way of asking about service history on VA.gov, so some example of what's meant by "services rendered" would be helpful.
   - [Step 3, Service periods](https://www.sketch.com/s/de782a35-e147-4c32-a2a8-ba53071ec8e7/a/j4DYVrb) - Some of the items required here (grade, rank, rating, organization) might not be information family members know or even know how to look up, depending on how familiar they are with the deceased Veteran's military service. An additional info component that provides some context or guidance on how to find this information could be helpful.

**Comment by fiorella-io** [comment]
@briandeconinck For the Should feedback, 
`Selecting "Yes" is what reveals the new field, which means users will have to tab past the "No" option before getting to the new field.` If we switch Yes and No could that fix this issue? Would switching yes /no break consistency on how yes/no options appear?
